examples/Microsoft.Spark.CSharp.Examples/Sql/Streaming/README.md

```diff
@@ -0,0 +1,130 @@
+
+#### Connect to Stream: ReadStream()
+
+The `ReadStream()` method returns a `DataStreamReader` that can be used to read streaming data in as a `DataFrame`. We'll include the host and port information so that our Spark app knows where to expect its streaming data.
+
+```CSharp
+DataFrame lines = spark
+    .ReadStream()
+    .Format("socket")
+    .Option("host", hostname)
+    .Option("port", port)
+    .Load();
+```
+
+### 3. Register a UDF
+
+A UDF is a *user-defined function.* We can use UDFs in Spark applications to perform calculations and analysis on our data.
+
+```CSharp
+Func<Column, Column> udfArray =
+    Udf<string, string[]>((str) => new string[] { str, $"{str} {str.Length}");
+```
+
+In the above code snippet from [StructuredNetworkCharacterCount.cs](StructuredNetworkCharacterCount.cs), we register a UDF called `udfArray`. This UDF will process each string it receives from the netcat terminal to produce an array that includes: the original string (contained in *str*), the original string concatenated with the length of that original string. 
+
+For example, entering *Hello world* in the terminal would produce an array where:
+* array[0] = Hello world
+* array[1] = Hello world 11
+    
+This is just an example of how you can use UDFs to further modify and analyze your data, even live as it's being streamed in!
+
+### 4. Use SparkSQL
+
+Next, we'll use SparkSQL to perform various functions on the data stored in our DataFrame. It's common to combine UDFs and SparkSQL so that we can apply a UDF to each row of our DataFrame.
+
+```CSharp
+DataFrame arrayDF = lines.Select(Explode(udfArray(lines["value"])));
+```
+
+In the above code snippet from [StructuredNetworkCharacterCount.cs](StructuredNetworkCharacterCount.cs), we apply *udfArray* to each value in our DataFrame (which represents each string read in from our netcat terminal). We then apply the SparkSQL method `Explode` to put each entry of our array in its own row. Finally, we use `Select` to place the columns we've produced in the new DataFrame *arrayDF.*
+
+### 5. Display Your Stream
+
+We can use `DataFrame.WriteStream()` to establish characteristics of our output, such as printing our results to the console and only displaying the most recent output and not all of our previous output as well. 
+
+```CSharp
+StreamingQuery query = arrayDf
+    .WriteStream()
+    .Format("console")
+    .Start();
+```
+
+### 6. Running Your Code
+
+Structured streaming in Spark processes data through a series of small **batches.** 
+When you run your program, the command prompt where we established the netcat will allow you to start typing.
+In our example, when you hit *enter* after entering data in the command prompt, Spark will consider that a batch and run the UDF. 
+
+![StreamingOutput](https://github.com/bamurtaugh/spark/blob/StreamingLog/examples/Microsoft.Spark.CSharp.Examples/Sql/Streaming/streamingnc.PNG)
```

Comment:
Can you update this image ?  The resulting table doesn't reflect what the `Udf` does.

Reply:
Updated, thanks for catching!